src/backends/perf_pt/decode.c

```diff
@@ -213,21 +215,94 @@ perf_pt_next_block(struct pt_block_decoder *decoder, int *decoder_status,
 }
 
 /*
- * Given a decoder and the last decoder status, drains events in the PT packet
- * stream.
+ * Given a decoder and pointer to the decoder status, handle any pending events in
+ * the PT packet stream and update the decoder status.
  *
- * The new decoder status is returned.
+ * Returns true on success, or false if an error occurred (e.g.) trace buffer
+ * overflow.
  */
-static int
-drain_events(struct pt_block_decoder *decoder, int decoder_status) {
-    while(decoder_status & pts_event_pending) {
+static bool
+handle_events(struct pt_block_decoder *decoder, int *decoder_status) {
+    bool ret = true;
+
+    while(*decoder_status & pts_event_pending) {
         struct pt_event event;
-        decoder_status = pt_blk_event(decoder, &event, sizeof(event));
-        if (decoder_status < 0) {
-            return decoder_status;
+        *decoder_status = pt_blk_event(decoder, &event, sizeof(event));
+        if (*decoder_status < 0) {
+            return *decoder_status;
+        }
+
+        switch (event.type) {
```

Comment:
Something's wrong with the indentation of the `case`: is it a tabs issue? They should be indented, right?

Reply:
It was actually intentional. I've seen all kinds of indent for switch statements. This one saves vertical space since we have large block comments. I can indent it once more if you feel strongly.

Reply:
I'm not saying it's a bad idea, but it is unusual, so I'm inclined to follow convention.